### ISSUE TITLE
Training with image input feature fails

### ISSUE THREAD

**Issue by SumanthDatta-Kony** [ludwig-ai/ludwig#4149]
With ludwig version 0.14.1  + ray , reading a dask dataframe  , one of the column has an image input feature fails with the error as shown in the attachment, it works fine with the previous ludwig versions. 0.10.x  Also the image input feature documentation is missing, please fix the below link. 

https://ludwig.ai/0.15/configuration/features/image_features/

I think the image binary data is being treated as a text.

below is the ludwig configuration 

{   'backend': {'trainer': {'num_workers': 7}, 'type': 'ray'},
    'input_features': [   {   'encoder': {   'fc_layers': [   {   'dropout': 0.4,
                                                                  'output_size': 256}],
                                             'num_fc_layers': 1},
                              'name': 'image_data',
                              'preprocessing': {'num_processes': 4},
                              'type': 'image'}],
    'ludwig_version': '0.14.1',
    'output_features': [{'name': 'reading', 'type': 'number'}],
    'trainer': {'early_stop': 10, 'epochs': 100}}

[ludwig014_image_training_error.txt](https://github.com/user-attachments/files/27419363/ludwig014_image_training_error.txt)

**Comment by SumanthDatta-Kony** [comment]
@w4nderlust , any suggestions to overcome this error.

**Comment by w4nderlust** [comment]
Thanks for the detailed report and the attached traceback, it helped us find the root cause.

**The bug was real and is fixed in the upcoming release.** Despite the `convert-string: False` setting existing in 0.14.1, it was applied inside `RayBackend.initialize()`, which runs *after* `model.train()` is called. By that point your Dask DataFrame already existed with the broken `_to_string_dtype` node baked into its task graph — too late.

**Fix:** We moved the `dask.config.set({"dataframe.convert-string": False})` call to `ludwig/__init__.py`, so it takes effect the instant you `import ludwig`, before any Dask DataFrame is created.
**Workaround for 0.14.1/0.15.x while you wait for the next release**, add this at the top of your script, *before* creating your Dask DataFrame:

```python
import dask
dask.config.set({"dataframe.convert-string": False})
import dask.dataframe as dd
# ... create your dataframe as usual
```

This one-liner exactly mimics what the fix does, so it will unblock you immediately.

**Comment by SumanthDatta-Kony** [comment]
@w4nderlust , Upgraded to ludwig 0.16.1  , above error is not seen , after "preprocessing is done"  stage .   During "initializing  the model" stage , fails with error "Distributed strategy not initialized". I think this is specific to image type columns, let me confirm.

PFA for the error .

[distribution_strategy_error_ludwig016.txt](https://github.com/user-attachments/files/27499079/distribution_strategy_error_ludwig016.txt)

**Comment by w4nderlust** [comment]
Thanks for your patience @SumanthDatta-Kony . Will look into it this eveninf

**Comment by w4nderlust** [comment]
Found it. `tune_batch_size_fn` and `tune_learning_rate_fn` are Ray remote tasks that run a full training step (to measure throughput), which triggers TorchMetrics' `sync_context()` → `get_current_dist_strategy()`. But unlike `train_fn`/`eval_fn`, these tuning workers never called `init_dist_strategy("local")` first — so the strategy was uninitialized and you got the `RuntimeError`.

Fix is in `main` now (commit `2f2c111`): both functions now call `init_dist_strategy("local")` immediately after `initialize_pytorch()`, exactly matching `train_fn`.

Install from `main` to get it now:
```bash
pip install git+https://github.com/ludwig-ai/ludwig.git@main
```

Will be included in the next patch release.

**Comment by SumanthDatta-Kony** [comment]
@w4nderlust , was it specific to the image type input feature? or more number of ray workers were included in the training config? Because i didn't see this in other trainings which didn't include image input feature.

**Comment by w4nderlust** [comment]
Neither, actually. The image feature was a red herring, the real discriminator was the **output feature type**.

The bug hit any config where the output feature uses a metric that goes through the full TorchMetrics `forward()` path (i.e. does NOT inherit `MeanMetric`). Your config had `type: number` output, whose `MSEMetric` hits `sync_context()`. A `category` output uses `SoftmaxCrossEntropyMetric(LossMetric(MeanMetric))`, `MeanMetric` subclasses take a shortcut in Ludwig's `base_feature.py` that bypasses TorchMetrics entirely, so `sync_context()` is never reached, and the bug is invisible.

The image input feature had nothing to do with it. You'd see the same crash with any input features as long as the output was `number`, `binary`, or anything else backed by a non-`MeanMetric` metric class.

It's fixed now though, with regression test to prevent it from resurfacing.
